controllers/hetznercluster_controller.go

```diff
@@ -409,6 +409,10 @@ func getAndValidateHCloudToken(ctx context.Context, namespace string, hetznerClu
 		return "", nil, err
 	}
 
+	if !hetznerSecret.DeletionTimestamp.IsZero() {
```

Comment:
I suppose this could be problematic  o?

Reply:
I removed that again. Afaik it is not needed any more. Now the root cause of the flaky test was fixed: Some tests created secrets without deleting them. This made the controller reconcile again and again.

Reply:
@janiskemper grrr after removing that line, the test failed again:
 
 > [FAIL] Hetzner ClusterReconciler cluster tests load balancer [It] should take over an existing load balancer with correct name

Reply:
solved now.